RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT_vect.cc

```diff
@@ -0,0 +1,1030 @@
+#include "RecoVertex/PrimaryVertexProducer/interface/DAClusterizerInZT_vect.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/GeometryCommonDetAlgo/interface/Measurement1D.h"
+#include "RecoVertex/VertexPrimitives/interface/VertexException.h"
+
+#include <cmath>
+#include <cassert>
+#include <limits>
+#include <iomanip>
+#include "FWCore/Utilities/interface/isFinite.h"
+#include "vdt/vdtMath.h"
+
+using namespace std;
+
+DAClusterizerInZT_vect::DAClusterizerInZT_vect(const edm::ParameterSet& conf) {
+
+  // hardcoded parameters
+  maxIterations_ = 100;
+  mintrkweight_ = 0.5; // conf.getParameter<double>("mintrkweight");
+
+
+  // configurable debug outptut debug output
+  verbose_ = conf.getUntrackedParameter<bool> ("verbose", false);
+  zdumpcenter_ = conf.getUntrackedParameter<double> ("zdumpcenter", 0.);
+  zdumpwidth_ = conf.getUntrackedParameter<double> ("zdumpwidth", 20.);
+  
+  // configurable parameters
+  double minT = conf.getParameter<double> ("Tmin")*std::sqrt(2.0);
+  double purgeT = conf.getParameter<double> ("Tpurge")*std::sqrt(2.0);
+  double stopT = conf.getParameter<double> ("Tstop")*std::sqrt(2.0);
```

Comment:
reposting my earlier comment left with no response

is there a good reason to have the magic sqrt(2) here instead of clearly having it in the values of parameters?

Reply:
The sqrt(2) would be sqrt(N) if this were templated in N dimensions.

Reply:
but why hide magic rescaling instead of having these parameters explicitly in the configuration?

Reply:
Because magic is best when you don't know the secret. ;-)